patches/components-update_client-updater_state_win.cc.patch

```diff
@@ -0,0 +1,35 @@
+diff --git a/components/update_client/updater_state_win.cc b/components/update_client/updater_state_win.cc
+index 66a049048b4b99b3296af25f0ffc914cf0ed8f2c..e5481826109b4089253e2d2cc390f1b79ce08a0b 100644
+--- a/components/update_client/updater_state_win.cc
++++ b/components/update_client/updater_state_win.cc
+@@ -23,6 +23,22 @@ namespace update_client {
+ 
+ namespace {
+ 
++#if defined(BRAVE_CHROMIUM_BUILD)
++const wchar_t kGoogleUpdatePoliciesKey[] =
++    L"SOFTWARE\\Policies\\BraveSoftware\\Update";
++const wchar_t kCheckPeriodOverrideMinutes[] = L"AutoUpdateCheckPeriodMinutes";
++const wchar_t kUpdatePolicyValue[] = L"UpdateDefault";
```

Comment:
most of these values look identical to the google versions. Seems like it would be better to use the undefine/redefine override. @bbondy can help with that

Reply:
const overriding should work just like method overriding. same def/undef applies

Reply:
why would the result be the same as copying if we're only changing some of the values?

Reply:
I referred `browser_about_handler.cc` in `chromium_src`.
In that file, upstream method name was changed and reused with additional implementation(`FixupBrowserAboutURLBraveImpl`).

I don't know how to adopt this technique to this file w/o changing functions implementation in this file.
Do you mean variable like re-define `kGoogleUpdatePoliciesKey` to `kGoogleUpdatePoliciesKey_ChromiumImpl`?

I think I don't know how to adopt undefine/redefine technique here.

Reply:
you really don't even have to redefine the old one here, I think you can just undef the old one and add a new one

Reply:
Do you mean like below?
```
#include "../../../../components/update_client/updater_state_win.cc"

#undef kGoogleUpdatePoliciesKey
#undef kCheckPeriodOverrideMinutes
#undef kUpdatePolicyValue
#undef kChromeUpdatePolicyOverride
#undef kRegPathGoogleUpdate 
#undef kRegPathClientsGoogleUpdate

namespace update_client {

namespace {

const wchar_t kGoogleUpdatePoliciesKey[] =
    L"SOFTWARE\\Policies\\BraveSoftware\\Update";
const wchar_t kCheckPeriodOverrideMinutes[] = L"AutoUpdateCheckPeriodMinutes";
const wchar_t kUpdatePolicyValue[] = L"UpdateDefault";
const wchar_t kChromeUpdatePolicyOverride[] =
    L"Update{AFE6A462-C574-4B8A-AF43-4CC60DF4563B}";
const wchar_t kRegPathGoogleUpdate[] = L"Software\\BraveSoftware\\Update";
const wchar_t kRegPathClientsGoogleUpdate[] =
    L"Software\\BraveSoftware\\Update\\Clients\\"
    L"{B131C935-9BE6-41DA-9599-1F776BEB8019}";

}  // namespace

}  // namespace update_client

```

Reply:
I did like this - https://github.com/brave/brave-core/pull/109/commits/b23256e3148924eb8ac3086aa97617fd3e4e47a8
Is this what you intended?

Reply:
does that work correctly? It doesn't seem like it would

Reply:
I didn't notice that they were inside an anonymous namespace and that might be problematic for undef. If that builds without error then I don't think it's using the new values

Reply:
but whether we patch or do something else, we only need the values we are overriding. We should not include things like kCheckPeriodOverrideMinutes or kCheckPeriodOverrideMinutesMax that we are not changing

Reply:
> does that work correctly? It doesn't seem like it would

It doesn't work

Reply:
Do you want to apply override technique like this (https://github.com/brave/brave-core/commit/b23256e3148924eb8ac3086aa97617fd3e4e47a8) to other files in this PR?

Reply:
well, it won't work if they're in an anonymous namespace like that. There's not way to reference/override them afaik